.github/workflows/main.yml

```diff
@@ -12,44 +12,62 @@ jobs:
-    - name: Install Python-Blosc  # Python-Blosc Sphinx conf.py imports blosc, not sure why
-      run: pip install -e $(pwd)/python-blosc/
-    - name: Install dependencies Python-Blosc2
-      run: pip install -r python-blosc2/requirements-doc.txt
-    - name: Pin pygments # Sphinx fails without this: 'HtmlFormatter' object has no attribute 'get_linenos_style_defs'
-      run: pip install pygments==2.11.2
+    - name: Set up environment and install dependencies
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        mamba-version: "*"
+        channels: conda-forge
+        activate-environment: blosc-docs
+        environment-file: doc/environment.yml
 
-    # Build websites
+    # blosc.org (nikola)
     - name: Build Nikola website
       uses: getnikola/nikola-action@v5
       with:
         dry_run: true
-    - name: Create artifact directory # The Nikola action creates output/ as root, and it's not writeable
+    - name: Create artifact directory
+      # The Nikola action creates `output/` as root, and it's not writeable
+      # `_site/` is the default directory for the upload/deploy pages actions
       run: mkdir _site && cp -r output/* _site/
+
+    # blosc.org/c-blosc2
+    - name: Check out C-Blosc2
+      run: git clone https://github.com/Blosc/c-blosc2.git
     - name: Build C-Blosc2 docs
-      run: sphinx-build -b html c-blosc2/doc/sphinx/source _site/c-blosc2
+      # TODO: Fix warnings in the build and restore `-W` parameter
+      run: sphinx-build c-blosc2/doc _site/c-blosc2
+      shell: bash -l {0}
+
+    # blosc.org/python-blosc
+    - name: Check out Python-Blosc
+      run: git clone https://github.com/Blosc/python-blosc.git
+    - name: Build Python-Blosc
+      run: cd python-blosc && python setup.py build  # python-blosc must be build from its directory (otherwise can't find README.rst used in setup.py)
+      shell: bash -l {0}
+    - name: Install Python-Blosc
+      run: pip install -e $(pwd)/python-blosc/
+      shell: bash -l {0}
     - name: Build Python-Blosc docs
-      run: sphinx-build -b html python-blosc/doc _site/python-blosc
+      # TODO: Fix warnings in the build and restore `-W` parameter
+      run: cd python-blosc && sphinx-build doc ../_site/python-blosc  # sphinx-build needs to be called from the docs directory, as some paths are relative
+      shell: bash -l {0}
+
+    # blosc.org/python-blosc2
+    - name: Check out Python-Blosc2
+      run: git clone --recurse-submodules https://github.com/Blosc/python-blosc2.git
+    - name: Build Python-Blosc2
+      run: cd python-blosc2 && python setup.py build_ext --inplace
+      shell: bash -l {0}
+    - name: Install Python-Blosc2
+      run: pip install -e $(pwd)/python-blosc2/
+      shell: bash -l {0}
     - name: Build Python-Blosc2 docs
-      run: sphinx-build -b html python-blosc2/doc _site/python-blosc2
+      run: cd python-blosc2 && sphinx-build -W doc ../_site/python-blosc2
```

Comment:
I am not an expert, but our canonical approach to build HTML docs is:

```
cd doc
PYTHONPATH=.. make html
```

and (after a lot of warnings that we should fix sooner rather than later) docs will appear in `_build/html/`

Reply:
`make html` just calls `sphinx-build`. I tried to set `PYTHONPATH`, but that shouldn't be needed since the source directory is installed with `pip install -e`.